parcels/field.py

```diff
@@ -826,6 +839,73 @@ def ccode_eval(self, var, t, z, y, x):
     def ccode_convert(self, _, z, y, x):
         return self.units.ccode_to_target(x, y, z)
 
+    #def find_in_block(self, chunksize, ti, zi, yi, xi):
+    #    #index = (ti, zi, yi, xi)
+    #    index = (ti, yi, xi)
+    #    block = tuple((int(index[i]/chunksize[i]) for i in range(len(index))))
+    #    local_index = tuple(index[i] - chunksize[i]*block[i] for i in range(len(index)))
+    #    return block, local_index
+
+    def get_block_id(self, block):
+        return np.ravel_multi_index(block, self.nchunks)
+
+    def get_block(self, bid):
+        return np.unravel_index(bid, self.nchunks)
+
+    def chunk_data(self):
+        if isinstance(self.data, da.core.Array):
+            if len(self.data.chunks[0]) > 1:
+                self.data = self.data.rechunk(sum(((self.grid.tdim,), self.data.chunksize[1:]), ()))
```

Comment:
Is there some kind of design documentation for this chunking scheme? I don't really understand the rechunking that goes on, particularly this rechunk to span the full time dimension. Maybe it makes sense given how `computeTimeChunk` operates. Do loaded chunks remain loaded, and are just stepped forward in time? It's a bit hard to piece this all together as a newcomer to the project.

Reply:
Hi @angus-g,

First of all, thank you for your constructive comments!
As you've already guessed, I was on holidays and didn't see your posts. I'm now back to business: I will take into account and answer your comments, thanks!

Reply:
I'll clean the rechunking and document it.

Something to point out already is the fact that we don't chunk the same way in space and time.
Indeed, while we take only the chunks that we need in space and let the rest being lazily open, we can't do that in time, basically because all the time steps are generally written on different files, and we don't want to open all files at once. So we manually control which files to open to have the correct time steps loaded. In space, we use the chunks as developed in this branch.

But, there is something wrong in dask and xarray (see https://github.com/pydata/xarray/issues/3041) such that files that should have been closed remain actually open.
That brings us a lot of problems, and explains while the chunking fails to be quick on fields with many time steps.

I'm now manually closing the files when I'm sure I won't use them anymore (I'll push that work somewhen next week)